src/main/java/com/iota/iri/model/Hash.java

```diff
@@ -19,33 +21,56 @@
     private byte[] bytes;
     private int[] trits;
```

Comment:
why not just make these 2 guys final, and use new instances whenever you need a Hash, nothing to put synchronized sections on, just make a new copy.

In my experience, synchronized should be a last resource. Try to work it functional, no mutability.
All those synchronized blocks most likely will end up creating deadlock headaches.